server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DomainMappingsRoutes.java

```diff
@@ -114,28 +126,56 @@ public HaltException removeDomainMapping(Request request, Response response) {
             @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
                     message = "Internal server error - Something went bad on the server side.")
     })
-    public Map<String, List<String>> get(Request request, Response response) throws RecipientRewriteTableException {
+    public Map<String, List<String>> getAllMappings(Request request, Response response) throws RecipientRewriteTableException {
         return recipientRewriteTable.getAllMappings()
                 .entrySet()
                 .stream()
-                .collect(toImmutableMap(e -> e.getKey().getFixedDomain(),
-                        e -> e.getValue()
-                                .select(Mapping.Type.Domain)
-                                .asStream()
-                                .map(Mapping::asString)
-                                .map(Mapping.Type.Domain::withoutPrefix)
-                                .collect(toImmutableList())
+                .filter(mappingsEntry -> !mappingsEntry.getValue().isEmpty())
+                .filter(mappingsEntry -> mappingsEntry.getValue().contains(Mapping.Type.Domain))
```

Comment:
The problem here is that we do not remove entries of the mappings that are not a domainMapping.

What I would suggest you to do it:

```
.map(mappingEntry -> Pair.of(mappingEntry.getKey(), mappingEntry.getValue().select(Mapping.Type.Domain))
.filter(mappingsEntry -> !mappingsEntry.getValue().isEmpty())
```

(We ensure to have only domain mappings, then we check it's not empty...).